packages/devextreme/js/__internal/ui/chat/chat.ts

```diff
@@ -98,27 +98,52 @@ class Chat extends Widget<Properties> {
       text,
     };
 
-    // @ts-expect-error
-    this.renderMessage(message, user);
+    this.renderMessage(message);
     // @ts-expect-error
     this._messageSendAction?.({ message, event });
   }
 
+  _oneMessageWasAddedToTheEnd(value: Message[], previousValue: Message[]): boolean {
+    const valueLength = value.length;
+    const previousValueLength = previousValue.length;
+
+    const lastValueItem = value[valueLength - 1];
```

Comment:
The function does not handle cases where either value or previousValue is empty. If either array is empty, accessing value[valueLength - 1] or previousValue[previousValueLength - 1] would result in an undefined value, which could lead to incorrect results or errors.

Reply:
[a6e60f3147d6506aee48ba6b26388f48fe0fe0c9]